src/components/Viewport.ts

```diff
@@ -256,23 +256,31 @@ export default class Viewport {
     state.isAdaptiveCached = false;
   }
 
-  public resize(): void {
+  public resize(isBeforeSync?: boolean): void {
```

Comment:
I thought you were using more methods in this function, so I recommended you to merge it rather than creating another one.
But looks like you're just using a few of them. In this case, I think it is more clear to separate methods.
I was expecting something like this:
```js
public resize(process): void {
  process.updateSize && this.updateSize();
  process.updateOriginalPanelPositions() && this.updateOriginalPanelPosition();
  ...
}
```